python-spec/src/somacore/options.py

```diff
@@ -177,5 +181,9 @@ class ResultOrder(enum.Enum):
     pa.ChunkedArray,
 ]
 """A single coordinate range for one dimension of a sparse ndarray."""
+
 SparseNDCoords = Sequence[SparseNDCoord]
 """A sequence of coordinate ranges for reading sparse ndarrays."""
+
+SpatialRegion = Union[Sequence[int], Sequence[float], shapely.GeometryType]
```

Comment:
```suggestion
SpatialRegion = Union[Sequence[int], Sequence[float], shapely.geometry.base.BaseGeometry]
```


I think this actually needs to be `shapely.geometry.base.BaseGeometry` because `shapely.box` returns a `shapely.Polygon` which is a subclass of `shapely.geometry.base.BaseGeometry` but not `shapely.GeometryType`.

```
def test_point_cloud_read_region_basic(tmp_path):
    uri = tmp_path.as_uri()

    schema = pa.schema([("x", pa.float64()), ("y", pa.float64())])

    with soma.PointCloud.create(uri, schema=schema) as ptc:
        pydict = {
            "soma_joinid": [1, 2, 3, 4, 5],
            "x": [10, 20, 30, 40, 50],
            "y": [4.1, 5.2, 6.3, 7.4, 8.5],
        }
        rb = pa.Table.from_pydict(pydict)
        ptc.write(rb)

    with soma.PointCloud.open(uri, "r") as ptc:
        # Must be a bounding box
        with pytest.raises(ValueError):
            ptc.read_region(region=[])

        read_region = ptc.read_region(region=None)
        print(read_region.data.concat())

        # read_region = ptc.read_region(region=[10, 50, 1, 10])
        # print(read_region.data.concat())

        region = shapely.box(10.0, 50.0, 1.0, 10.0)
        print(type(region)) # <class 'shapely.geometry.polygon.Polygon'>
        print(isinstance(region, shapely.Polygon)) # True
        print(isinstance(region, shapely.GeometryType)) # False
        print(issubclass(shapely.Polygon, shapely.GeometryType)) # False
        print(isinstance(region, shapely.geometry.base.BaseGeometry)) # True
        print(issubclass(shapely.Polygon, shapely.geometry.base.BaseGeometry)) # True

        # >       raise TypeCheckError(f"did not match any element in the union:\n{formatted_errors}")
        # E       typeguard.TypeCheckError: argument "region" (shapely.geometry.polygon.Polygon) did not match any element in the union:
        # E         Sequence[int]: is not a sequence
        # E         Sequence[float]: is not a sequence
        # E         shapely._geometry.GeometryType: is not an instance of shapely._geometry.GeometryType
        # E         NoneType: is not an instance of NoneType
        read_region = ptc.read_region(region=region)
```